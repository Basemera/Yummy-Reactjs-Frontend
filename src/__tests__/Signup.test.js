import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson, { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';

import Signup from '../components/auth/signup';

describe('Signup component', () => {
  const wrapper = shallow(<Signup />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('has a form', () => {
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('form').simulate('submit', {preventDefault}))
    expect(preventDefault).toBeCalled();
    // expect(wrapper.find('[name="email_field"]').simulate('change', {target:{name:'email_field', value:'had@hmail.com'}}))
  });
  it('renders h1', () => {
    expect(wrapper.find('h1').length).toBe(1);
  });

  it('renders inputs', () => {
    expect(wrapper.find('input').length).toBe(4);
  });

  it('it has br', () => {
    expect(wrapper.find('br').length).toBe(2);
  });

  it('renders component in div', () => {
    expect(wrapper.find('div').length).toBe(8);
  });

  it('renders component in span', () => {
    expect(wrapper.find('span').length).toBe(2);
  });

  it('has the correct form fields', () => {
    expect(wrapper.find('#emailsignup')).toHaveLength(1);
    expect(wrapper.find('[name="password_field"]')).toHaveLength(1);
    expect(wrapper.find('#signup')).toHaveLength(1);
  });

  it('has empty form fields', () => {
    expect(wrapper.find('#emailsignup').props().value).toEqual('');
    expect(wrapper.find('[name="password_field"]').props().value).toEqual('');
    expect(wrapper.find('#signup').props().value).toEqual('Signup');
  });

  it('Form fields update when state changes', () => {
    wrapper.setState({ password_field: 'hi', email_field: 'mail' });
    expect(wrapper.find('#emailsignup').props().value).toEqual('mail');
    expect(wrapper.find('[name="password_field"]').props().value).toEqual('hi');
  });
});
