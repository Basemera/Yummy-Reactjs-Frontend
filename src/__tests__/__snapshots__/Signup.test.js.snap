// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup component renders properly 1`] = `
<div
  className="background"
>
  <div
    id="signbar"
  >
    <h1>
      CREATE AN ACCOUNT 
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="input-group"
      >
        <span
          className="input-group-addon pr-4"
          id="email"
        >
          <i
            className="fa fa-envelope"
          />
        </span>
        <input
          aria-describedby="email"
          className="form-control"
          id="emailsignup"
          name="email_field"
          onChange={[Function]}
          placeholder="example@gmail.com"
          value=""
        />
      </div>
      <br />
      <div
        className="input-group"
      >
        <span
          className="input-group-addon pr-4"
          id="email"
        >
          <i
            className="fa fa-lock"
          />
        </span>
        <input
          className="form-control"
          name="password_field"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </div>
      <br />
      <div
        className="row"
      >
        <div
          className="col-sm-6 text-left"
        >
          <Link
            replace={false}
            to="/"
          >
            <input
              className="btn-btn-default"
              type="button"
              value="Cancel"
            />
          </Link>
        </div>
        <div
          className="col-sm-6 text-right"
        >
          <input
            className="btn btn-nav"
            id="signup"
            name="signup"
            type="submit"
            value="Signup"
          />
        </div>
      </div>
      <div
        className="change_link"
      >
        Already a member ?
        <Link
          replace={false}
          to="/login"
        >
          log in
        </Link>
      </div>
    </form>
  </div>
</div>
`;
